Source/bmalloc/bmalloc/bmalloc.cpp

```diff
@@ -52,7 +53,10 @@ pas_primitive_heap_ref gigacageHeaps[static_cast<size_t>(Gigacage::NumberOfKinds
 
 void* mallocOutOfLine(size_t size, CompactAllocationMode mode, HeapKind kind)
 {
-    return malloc(size, mode, kind);
+    void* allocation = malloc(size, mode, kind);
+    if (mode == CompactAllocationMode::NonCompact && MAR_isAddressInQualifyingPage(allocation))
+        MAR_didAllocate(&marRegistry, allocation, size);
```

Comment:
I think this form will break the optimization of bmalloc's allocation because this is a non-inlined call after the tail-call form, isn't it correct? (Ensure that there is no PAC signing prologue at `fastMalloc` function etc.).